File: models/lead.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class LeadPushRequest(BaseModel):
    """
    Push lead slot state to Make.com → svlbh-v2.
    status: ACTIVE | WAITING | COMPLETED
    tier: LEAD | FORMATION | CERTIFIEE | SUPERVISEUR
    Known codes: Cornelia=300, Flavia=301, Anne=302, Patrick=455000
    """
    model_config = ConfigDict(populate_by_name=True)

    shamane_code: str = Field(..., alias="shamaneCode")
    prenom: str
    nom: Optional[str] = Field(None)
    tier: str = Field("CERTIFIEE")
    status: str = Field("ACTIVE")
    session_key: Optional[str] = Field(None, alias="sessionKey")
    platform: str = Field("android")


class LeadPullRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    shamane_code: str = Field(..., alias="shamaneCode")


class LeadPullResponse(BaseModel):
    shamane_code: str
    prenom: Optional[str] = None
    nom: Optional[str] = None
    tier: Optional[str] = None
    status: Optional[str] = None
    session_key: Optional[str] = None
    found: bool = False
    timestamp: Optional[int] = None

File: android/vlbh-energy-sdk/src/main/java/energy/vlbh/sdk/models/Lead.kt
package energy.vlbh.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LeadPushRequest(
    val shamaneCode: String,
    val prenom: String,
    val nom: String? = null,
    val tier: String = "CERTIFIEE",
    val status: String = "ACTIVE",
    val sessionKey: String? = null,
    val platform: String = "android"
)

@Serializable
data class LeadPullRequest(
    val shamaneCode: String
)

@Serializable
data class LeadPullResponse(
    @SerialName("shamane_code") val shamaneCode: String,
    val prenom: String? = null,
    val nom: String? = null,
    val tier: String? = null,
    val status: String? = null,
    @SerialName("session_key") val sessionKey: String? = null,
    val found: Boolean,
    val timestamp: Long? = null
)
